pkg/agent/core/ngt/service/kvs/kvs_test.go

```diff
@@ -295,57 +400,120 @@ func Test_bidi_Set(t *testing.T) {
+	type want struct {
+		key string
+		val uint32
+		l   uint64
+	}
 	type test struct {
 		name       string
 		args       args
 		fields     fields
 		want       want
-		checkFunc  func(want) error
+		checkFunc  func(w want, args args, b *bidi) error
 		beforeFunc func(args)
 		afterFunc  func(args)
 	}
-	defaultCheckFunc := func(w want) error {
+	defaultCheckFunc := func(w want, args args, b *bidi) error {
+		val, ok := b.Get(args.key)
+		if !ok {
+			return errors.New("val not found")
+		}
+		key, ok := b.GetInverse(args.val)
+		if !ok {
+			return errors.New("key not found")
+		}
+		if val != w.val {
+			return errors.Errorf("val is not equals.\twant: %v, but got: %v", w.val, val)
+		}
+		if key != w.key {
+			return errors.Errorf("key is not equals.\twant: %v, but got: %v", w.key, key)
+		}
+		if l := atomic.LoadUint64(&b.l); l != w.l {
+			return errors.Errorf("l is not equals.\twant: %v, but got: %v", l, w.l)
+		}
 		return nil
 	}
 	tests := []test{
-		// TODO test cases
-		/*
-		   {
-		       name: "test_case_1",
-		       args: args {
-		           key: "",
-		           val: 0,
-		       },
-		       fields: fields {
-		           ou: nil,
-		           uo: nil,
-		           l: 0,
-		       },
-		       want: want{},
-		       checkFunc: defaultCheckFunc,
-		   },
-		*/
-
-		// TODO test cases
-		/*
-		   func() test {
-		       return test {
-		           name: "test_case_2",
-		           args: args {
-		           key: "",
-		           val: 0,
-		           },
-		           fields: fields {
-		           ou: nil,
-		           uo: nil,
-		           l: 0,
-		           },
-		           want: want{},
-		           checkFunc: defaultCheckFunc,
-		       }
-		   }(),
-		*/
+		func() test {
+			fields := fields{
+				l: 0,
+			}
+			for i := 0; i < slen; i++ {
+				fields.ou[i] = new(ou)
+				fields.uo[i] = new(uo)
+			}
+
+			var (
+				key        = "45637ec4-c85f-11ea-87d0"
+				val uint32 = 14438
+			)
+
+			return test{
+				name: "set success",
+				args: args{
+					key: key,
+					val: val,
+				},
+				fields: fields,
+				want: want{
+					key: key,
+					val: val,
+					l:   1,
+				},
+			}
+		}(),
+		func() test {
+			fields := fields{
+				l: 0,
+			}
+			for i := 0; i < slen; i++ {
+				fields.ou[i] = new(ou)
+				fields.uo[i] = new(uo)
+			}
+
+			var (
+				key = "45637ec4-c85f-11ea-87d0"
+			)
+
+			return test{
+				name: "set success when the val is default value",
+				args: args{
+					key: key,
+				},
+				fields: fields,
+				want: want{
+					key: key,
+					val: 0,
+					l:   1,
+				},
+			}
+		}(),
+		func() test {
+			fields := fields{
+				l: 0,
+			}
+			for i := 0; i < slen; i++ {
+				fields.ou[i] = new(ou)
+				fields.uo[i] = new(uo)
+			}
+
+			var (
+				val uint32 = 14438
+			)
+
+			return test{
+				name: "set success when the key is default value",
+				args: args{
+					val: val,
+				},
+				fields: fields,
+				want: want{
+					val: val,
+					l:   1,
+				},
+			}
+		}(),
```

Comment:
can you add a test case that both key and val is 0/empty value?

Reply:
thank you
I fixed: https://github.com/vdaas/vald/pull/1223/commits/99c5dfc0c64934a9267dd8cee64732514242de36